netbox/dcim/forms/object_create.py

```diff
@@ -52,6 +52,17 @@ class ComponentCreateForm(forms.Form):
     # ComponentCreateView when creating objects.
     replication_fields = ('name', 'label')
 
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        # Components attached to a module need to present this standardized substitution help text.
+        if 'module' in self.fields:
```

Comment:
This doesn't address the reproduction steps in #17459, which entails a module type (as opposed to a module). Both fields should get the same help text.

Reply:
Centralized the help text string in `common.py` (maybe there is a better place for this?) and reused both in `ModularComponentTemplateForm` (which is used in adding component templates to module types) and `ComponentCreateForm` (which is used when creating components directly).

Reply:
I misspoke earlier; only module _types_ support the use of the `{module}` token in component names. (This is replaced with the position of the module being created.) It is not supported for components being created on existing modules.

Reply:
From what I can understand from the documentation, the text about `{module}` should ONLY appear on the component _template_ create forms (derived from `ModularComponentTemplateForm`), concatenated with the range expansion text. This applies to both workflows where component templates are created — when attaching to `ModuleType`s and to `DeviceType`s.

The `{module}` text should NOT appear on the direct object creation forms at all, including `InterfaceCreateForm` where it was appearing prior to this PR.

Is that correct?